### ISSUE TITLE
Unable to initialize BooleanField to it's checked state

### ISSUE THREAD

**Issue by Bob64375** [wtforms/wtforms#542]
When obj is used to initialize a form (See form.py):

```
def process(self, formdata=None, obj=None, data=None, **kwargs):
   :param obj:
            If formdata is empty or not provided, this object is checked for
            attributes matching form field names, which will be used for field
            values.
```

A true initial value in obj for a Boolean field (the intent is for the field to be rendered as checked) is incorrectly reset to false by process_formdata and the rendered result is incorrectly not-checked. Please see the process_formdata method in class BooleanField located in fields/core.py for this code:

```python
 def process_formdata(self, valuelist):
        if not valuelist or valuelist[0] in self.false_values:
            self.data = False
        else:
            self.data = True
```

IMO it should be something like this. Without testing for self.data (as in the above code) the BooleanField cannot be set to true (checked) :

```python
def process_formdata(self, valuelist):
        if valuelist:
            if valuelist[0] in self.false_values:
                self.data = False
            else:
                self.data = True
        elif self.data is None:
            self.data = False
```

**Comment by azmeuk** [comment]
It seems to be related to #460. Can you confirm this?

**Comment by Bob64375** [comment]
Possibly, but I don’t have the time to dig into it, sorry!

**Comment by azmeuk** [comment]
Then I will close this issue for now as I believe it is a duplicate. Do not hesitate to tell if you think the issues are different.

**Comment by Bob64375** [comment]
I would not recommend closing #542 <https://github.com/wtforms/wtforms/issues/542> .

 

#460 <https://github.com/wtforms/wtforms/issues/460>  and #542 <https://github.com/wtforms/wtforms/issues/542>  are very different in that #460 is providing test cases in which BooleanField does the wrong thing, whereas #542 shows broken code in the WTForms BooleanField class and offers a fix. 

 

I recommend a WTForms developer review the fix provided in #542 and commit it to the repository. The review should include some discussion of fixing the BooleanField class in that the fix could change behavior of sites that rely on BooleanField to be broken.

 

I am neither a WTForms developer not sufficiently connected to have that discussion.

 

It would be easy for a WTForms developer to commit the fix provided in #542 to the repository and fix the broken BooleanField!
